tls/s2n_security_policies.c

```diff
@@ -621,6 +628,12 @@ int s2n_connection_set_cipher_preferences(struct s2n_connection *conn, const cha
     notnull_check(&conn->security_policy_override->kem_preferences);
     notnull_check(&conn->security_policy_override->signature_preferences);
     notnull_check(&conn->security_policy_override->ecc_preferences);
+
+    if (s2n_is_in_fips_mode()) {
+        ENSURE_POSIX(conn->security_policy_override->kem_preferences == &kem_preferences_null,
+                S2N_ERR_PQ_KEMS_DISALLOWED_IN_FIPS);
+    }
+
```

Comment:
Same comment, pass a temp variable to `s2n_find_security_policy_from_version()`, and only set `conn->security_policy_override` after all the checks have passed.